plugins/net_plugin/net_plugin.cpp

```diff
@@ -2633,6 +2628,12 @@ namespace eosio {
          fc::optional<transaction_id_type> trx_id;
          fc::raw::unpack( ds, trx_id );
          if( trx_id ) {
+            if (trx_in_progress_sz > def_max_trx_in_progress_size) {
+               char reason[72];
+               snprintf(reason, 72, "Dropping trx, too many trx in progress %lu bytes", trx_in_progress_sz);
+               my_impl->producer_plug->log_failed_transaction(*trx_id, reason);
+               return true;
+            }
```

Comment:
also, at line 2646, change EOS_ASSERT(...) to:
if (trx_id && ptr && *trx_id != ptr->id()) {
   ...->log_failed_trasaction(ptr->id(), "... Provided trx_id does not ..."
  EOS_ASSERT(false,...)
}
   
}